spec/unit/agent_spec.rb

```diff
@@ -198,18 +197,17 @@ def controlled_run(&block)
         expect(client).to receive(:run)
 
         expect(Kernel).to receive(:fork).and_yield
-        @agent.run
+        expect { @agent.run }.to exit_with(1)
```

Comment:
Normally, I would expect this to `exit_with(0)` in the "happy path". But I think it exits with 1, because the [`AgentTestClient#run` method](https://github.com/puppetlabs/puppet/pull/8074/files#diff-4182e2d80f6f6fd20e3fac1bcf36dbe0L7) returns `nil` implicitly, triggering the `exit(yield || 1)` behavior. So I think this test is correct though surprising at first glance.

Reply:
I think is better to have `AgentTestClient#run` return 0 and test `exit` with 0 in this test